server_src/constants/state.constants.js

```diff
@@ -0,0 +1,222 @@
+const getWolPowerSubSettingsDefault = () => {
+  return {
+    enabled: false,
+    ip: "255.255.255.0",
+    packets: "3",
+    port: "9",
+    interval: "100",
+    MAC: ""
+  };
+};
+
+const getPowerSettingsDefault = () => {
+  return {
+    powerOnCommand: "",
+    powerOnURL: "",
+    powerOffCommand: "",
+    powerOffURL: "",
+    powerToggleCommand: "",
+    powerToggleURL: "",
+    powerStatusCommand: "",
+    powerStatusURL: "",
+    wol: getWolPowerSubSettingsDefault()
+  };
+};
+
+const getCostSettingsDefault = () => {
+  return {
+    powerConsumption: 0.5,
+    electricityCosts: 0.15,
+    purchasePrice: 500,
+    estimateLifespan: 43800,
+    maintenanceCosts: 0.25
+  };
+};
+
+const getFilterDefaults = () => [
+  "All Printers",
+  "State: Idle",
+  "State: Active",
+  "State: Complete",
+  "State: Disconnected"
+];
+
+const SYSTEM_CHECKS = {
+  api: "api",
+  files: "files",
+  state: "state",
+  profile: "profile",
+  settings: "settings",
+  system: "system"
+
+  //Special
+  // cleaning: "cleaning"
+};
+
+const getSystemChecksDefault = () => {
+  return {
+    [SYSTEM_CHECKS.api]: {
+      status: "danger",
+      date: null
+    },
+    [SYSTEM_CHECKS.files]: {
+      status: "danger",
+      date: null
+    },
+    [SYSTEM_CHECKS.state]: {
+      status: "danger",
+      date: null
+    },
+    [SYSTEM_CHECKS.profile]: {
+      status: "danger",
+      date: null
+    },
+    [SYSTEM_CHECKS.settings]: {
+      status: "danger",
+      date: null
+    },
+    [SYSTEM_CHECKS.system]: {
+      status: "danger",
+      date: null
+    },
+    // TODO Hmm
+    cleaning: {
+      information: {
+        date: null
+      },
+      job: {
+        date: null
+      },
+      file: {
+        date: null
+      },
+      status: "danger",
+      date: null
```

Comment:
I like the idea for this, it seems more of a OctoFarm thing than a printer thing personally.

Reply:
That's a good point. Im actually struggling with it in the new architecture. The systemChecks are coupled to each printer on `master`
https://github.com/OctoFarm/OctoFarm/blob/e99cabd3f27f3ab6e21c27f0da5e8e319c4a2751/server_src/lib/dataFunctions/printerClean.js#L596
So in that regard it used to be printer based.... or let me say "device based" if we're abstracting to multiple interfaces

Reply:
Yeah sorry this was more of a " should we move this ? " question than a statement. As these are invariably going to to be task based now no? 

The UI could be on System as a little health check and the last run time outputted on the log dump (just as ideas)

Reply:
My mind is a bit fried... Ive been struggling with how we process data and my mental solution is to not do it backend but on client side. Im want to make sure that it's not because of my bad design so Ill ponder on it a bit

Reply:
> Answering your question: Im not sure it should be task based as updates are mostly caused by WebSocket messages. I would expect the resulting data from that to be updated by websocket state changes

Job and Printer yeah totally websockets true. 

History and other already in there aha. 

> my mental solution is to not do it backend but on client side. Im want to make sure that it's not because of my bad design so Ill ponder on it a bit.
To me the UI shoud UI for lack of a better description aha. So anything relating to displaying information. The server should be prepping information, ie calculations/settings and what not. There is a rather blatant dis-regard for that at the moment with some html server side, which should go.